R/classes.R

```diff
@@ -150,47 +150,72 @@ create_process_object <- function(res) {
   }
 
   # assign class and return
-  class(res) <- c("babylon_process", class(res))
+  class(res) <- c(PROC_CLASS, class(res))
   return(res)
 }
 
-#' @describeIn create_bbi_object Create list object of `bbi_run_log_df` class, first checking that all the required columns are present.
-#' @param log_df data.frame or tibble to attempt to assign the class to
+#' @describeIn create_bbi_object Create tibble object of `bbi_run_log_df` class, first checking that all the required columns are present.
 #' @keywords internal
 create_run_log_object <- function(log_df) {
+  log_df <- create_log_df_impl(log_df, RUN_LOG_CLASS, RUN_LOG_REQ_COLS, ABS_MOD_PATH)
+  return(log_df)
+}
+
+#' @describeIn create_bbi_object Create tibble object of `bbi_config_log_df` class, first checking that all the required columns are present.
+#' @keywords internal
+create_config_log_object <- function(log_df) {
+  log_df <- create_log_df_impl(log_df, CONF_LOG_CLASS, CONFIG_LOG_REQ_COLS, ABS_MOD_PATH)
+  return(log_df)
+}
+
+#' @describeIn create_bbi_object Create tibble object of `bbi_summary_log_df` class, first checking that all the required columns are present.
+#' @keywords internal
+create_summary_log_object <- function(log_df) {
```

Comment:
Are all of the `create_*_log_object()` functions necessary to implement the new functionality for `param_estimates()` (as described in #113)? I can see the case for `create_summary_log_object()`, and the desire to have a unified approach to creating such objects, but I'm not sure the other functions are strictly necessary. If this is the case, can they be moved into a separate PR that focuses just on unifying `*_log_object` creation?

Reply:
They are not strictly related, they're a refactor that happened naturally as part of development. I was trying to dispatch on some classes and then realized those classes didn't exist and then realized there was no standard way to create them consistently. So I created one.